src/main/java/edu/ie3/datamodel/io/source/csv/CsvDataSource.java

```diff
@@ -168,6 +168,14 @@ protected Map<String, String> buildFieldsToAttributes(
 
     try {
       String[] fieldVals = parseCsvRow(csvRow, csvSep);
+
+      if (fieldVals.length > headline.length) {
+        throw new SourceException(
+            "The size of the fields is greater than the size of the headline. This can happen when using the old"
+                + " and now unsupported csv input format! Please refer to the docs to find the way to convert the"
+                + " input data into the new input format.");
+      }
+
       insensitiveFieldsToAttributes.putAll(
           IntStream.range(0, fieldVals.length)
```

Comment:
I think if do this here, we don't need the additional if clause above. The error check below should cover it pretty well.

```suggestion
          IntStream.range(0, Math.min(fieldVals.length, headline.length))
```